synchronicity/utils.py

```diff
@@ -0,0 +1,27 @@
+import functools
+import sys
+import traceback
+
+def get_tb_filename(tb):
+    return traceback.extract_tb(tb, 1)[0].filename
+
+
+def get_filtered_tb(tb):
+    while tb and "synchronicity" in get_tb_filename(tb):
```

Comment:
1. Feels a bit ugly to use string matching here. Can compare the filename to see if it's in the same directory as `__file__` maybe?
2. Instead of a while-loop where you use `extract_tb(tb, limit=1)`, can you just extract the whole stack frame in one go (`limit=None`) and iterate over it?

Reply:
1. Yeah I figured there's a better way to do this. I didn't think __file__ was always set, but let me look into it. 
2. `extract_tb` returns a StackSummary object that's "processed" and no longer seems to have pointers. It would be ideal if we could filter the StackSummary list and then convert it back to a traceback object, but afaict there's no way to go from StackSummary -> traceback. So we still need to operate on the traceback objects and rewrite the pointers.

Reply:
`__file__` is always set inside whatever file you're in